phpunit/class-wp-get-block-css-selectors-test.php

```diff
@@ -0,0 +1,319 @@
+<?php
+/**
+ * Tests_Get_Block_CSS_Selector class
+ *
+ * @package WordPress
+ */
+
+class WP_Get_Block_CSS_Selector_Test extends WP_UnitTestCase {
+	private $test_block_name;
+
+	public function set_up() {
+		parent::set_up();
+		$this->test_block_name = null;
+	}
+
+	public function tear_down() {
+		unregister_block_type( $this->test_block_name );
+		$this->test_block_name = null;
+		parent::tear_down();
+	}
+
+	private function register_test_block( $name, $selectors = null, $supports = null ) {
+		$this->test_block_name = $name;
+
+		return register_block_type(
+			$this->test_block_name,
+			array(
+				'api_version' => 2,
+				'attributes'  => array(),
+				'selectors'   => $selectors,
+				'supports'    => $supports,
+			)
+		);
+	}
+
+	public function test_get_root_selector_via_selectors_api() {
+		$block_type = self::register_test_block(
+			'test/block-with-selectors',
+			array( 'root' => '.wp-custom-block-class' )
+		);
+
+		$selector = wp_get_block_css_selector( $block_type );
+		$this->assertEquals( '.wp-custom-block-class', $selector );
+	}
+
+	public function test_get_root_selector_via_experimental_property() {
+		$block_type = self::register_test_block(
+			'test/block-without-selectors',
+			null,
+			array( '__experimentalSelector' => '.experimental-selector' )
+		);
+
+		$selector = wp_get_block_css_selector( $block_type );
+		$this->assertEquals( '.experimental-selector', $selector );
+	}
+
+	public function test_default_root_selector_generation() {
```

Comment:
For completeness, it'd be great to also cover the block `core/without-selectors-or-supports`, as the generated selector uses a logic different from `test/without-selectors-or-supports`.

Reply:
✅ Done, thanks. https://github.com/WordPress/gutenberg/pull/46496/commits/bbb754f6183bf39f5a81ff79e99aa1e86430a770